### ISSUE TITLE
exclude datasets from certain variables

### ISSUE THREAD

**Issue by dhohn** [ESMValGroup/ESMValCore#1641]
I would like to be able to better control which datasets are ran for which variable. The reason being that Im looking at several variables as well as datasets and it happens frequently that one variable is not available from one particular model. It also happens often that iris or the cmor checker is not happy with one of the lot (and no quick fix exists either).

- What I did so far is (un)comment the datasets depending on which variable Im looking at, but that obviously gets old real quick.
- Then I thought about removing the global dataset list and have `additional_datasets` for each variable, but that includes a lot of repetition.
- Then I thought yaml anchors would come in handy and I could define partial dataset lists and then combine them as needed. Alas sequence merging is not yet supported (https://github.com/yaml/yaml/issues/48).
- Finally I considered adding a new keyword `excluded_datasets` so that I can selectively remove datasets and repeat myself *a bit* less.

**Would you be able to help out?**
Ive drafted an implementation. The new `excluded_datasets` keyword takes the same format and is recognised in the same places (ie. `variables` and `diagnostic`) as `additional_datasets`. The patch is attached. Let me know if this is of general enough interest for a PR.

[exclude.txt](https://github.com/ESMValGroup/ESMValCore/files/8959137/exclude.txt)

**Comment by dhohn** [comment]
Ah indeed: https://github.com/ESMValGroup/ESMValCore/issues/1538
Sorry I missed that!

I think it only becomes complicated if one starts mixing `additional_datasets` and `exclude_datasets` at the different levels. And probably there we would have issues with deciding and defining the order of precedence.

I do think it can make the intent more obvious when one usually wants to analyse *all* datasets at recipe level, but have a few exceptions for technical or availability reasons or whatever.

**Comment by bouweandela** [comment]
> I think it only becomes complicated if one starts mixing additional_datasets and exclude_datasets at the different levels. And probably there we would have issues with deciding and defining the order of precedence.


Yes, this is why I'm not too keen on the feature. I would like to simplify instead of complicate, therefore I also proposed to drop one of the datasets sections at some point, but there seemed to be no consensus: #92. Maybe we could think more along the lines of merging the sequences ourselves instead of waiting for YAML to do it for us?

**Comment by bouweandela** [comment]
Something like this?
```yaml
datasets:
  some_datasets:
    - {dataset: dataset1, ...}
    - {dataset: dataset2, ...}
  other_datasets:
    - {dataset: dataset3, ...}

diagnostics:
  diagnostic1:
    variables:
      var1:
        datasets: [some_datasets, other_datasets]
      var2:
        datasets: [some_datasets]
```

I'm not sure if this would be something that could go into a v2 release, we might have to wait for v3, whenever that is.

**Comment by bouweandela** [comment]
Anyway, so long as it is backward compatible people are probably fine with it, but having two different systems is of course also more complicated..

**Comment by dhohn** [comment]
I also like this suggestion (https://github.com/ESMValGroup/ESMValCore/issues/1641#issuecomment-1165532363), but I imagine quite a bit more work.

These are essentially dataset groups which could also be nice to use semantically, e.g. to have an alias for all subexperiments of one model or vice versa.
